src/System/Random/Internal.hs

```diff
@@ -709,34 +713,47 @@ instance UniformRange Bool where
 -- | See /Floating point number caveats/ in "System.Random.Stateful".
 instance UniformRange Double where
   uniformRM (l, h) g = do
-    w64 <- uniformWord64 g
-    let x = word64ToDoubleInUnitInterval w64
+    x <- uniformDouble01M g
     return $ (h - l) * x + l
 
--- | Turns a given uniformly distributed 'Word64' value into a uniformly
--- distributed 'Double' value in the range [0, 1].
-word64ToDoubleInUnitInterval :: Word64 -> Double
-word64ToDoubleInUnitInterval w64 = d / m
+-- | Generate uniformly distributed 'Double' in the range [0, 1].
+uniformDouble01M :: StatefulGen g m => g -> m Double
+uniformDouble01M g = do
+  w64 <- uniformWord64 g
+  return $ fromIntegral w64 / m
   where
-    d = fromIntegral w64 :: Double
     m = fromIntegral (maxBound :: Word64) :: Double
-{-# INLINE word64ToDoubleInUnitInterval #-}
+
+-- | Generate uniformly distributed 'Double' in the range (0, 1]. That
+--   is result is guaranteed to be positive
```

Comment:
I think that "That is result is guaranteed to be positive" can be phrased better or removed. Assuming the reader knows what `(0,1]` means, it is clear that the range is positive. So my first suggestion would be to leave it at that:

```suggestion
-- | Generates uniformly distributed 'Double' in the range (0, 1].
```
Alternatively, to drive home the point that this range is positive, we could insert the word "positive":

```suggestion
-- | Generates uniformly distributed 'Double' in the positive range (0, 1].
```

What do you think?